wagtail/images/__init__.py

```diff
@@ -39,3 +42,15 @@ def get_image_model():
             "WAGTAILIMAGES_IMAGE_MODEL refers to model '%s' that has not been installed"
             % model_string
         )
+
+
+def get_rendition_storage():
+    """
+    Obtain the storage object for an image rendition file.
+    Returns custom storage (if defined), or the default storage.
+    """
+    storage = getattr(settings, "WAGTAILIMAGES_RENDITION_STORAGE", default_storage)
+    if isinstance(storage, str):
+        import_module(storage)
+        storage = storage()
+    return storage
```

Comment:
I recommend we follow the existing pattern set up by `get_rendition_upload_to` and `get_upload_to`

So that it will still call a method on the rendition instance.

This provides the ability to globally change this via the setting or on a per instance change this if really needed.

Reply:
The problem with using this pattern is that when models are loaded at startup, there is no instance, which prevents the ImageField from initializing, causing an error.  If there's a lazy way to do this, it may be possible.  Still investigating.

`(wgtail310) eevel@weezel3:~/wagtail$ python runtests.py 
Traceback (most recent call last):
  File "/home/eevel/wagtail/runtests.py", line 102, in <module>
    runtests()
  File "/home/eevel/wagtail/runtests.py", line 93, in runtests
    execute_from_command_line(argv)
  File "/home/eevel/wgtail310/lib/python3.10/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
    utility.execute()
  File "/home/eevel/wgtail310/lib/python3.10/site-packages/django/core/management/__init__.py", line 420, in execute
    django.setup()
  File "/home/eevel/wgtail310/lib/python3.10/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/eevel/wgtail310/lib/python3.10/site-packages/django/apps/registry.py", line 116, in populate
    app_config.import_models()
  File "/home/eevel/wgtail310/lib/python3.10/site-packages/django/apps/config.py", line 304, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/eevel/wagtail/wagtail/test/testapp/models.py", line 54, in <module>
    from wagtail.images.blocks import ImageChooserBlock
  File "/home/eevel/wagtail/wagtail/images/blocks.py", line 7, in <module>
    from .shortcuts import get_rendition_or_not_found
  File "/home/eevel/wagtail/wagtail/images/shortcuts.py", line 1, in <module>
    from wagtail.images.models import SourceImageIOError
  File "/home/eevel/wagtail/wagtail/images/models.py", line 611, in <module>
    class AbstractRendition(ImageFileMixin, models.Model):
  File "/home/eevel/wagtail/wagtail/images/models.py", line 613, in AbstractRendition
    file = models.ImageField(
  File "/home/eevel/wgtail310/lib/python3.10/site-packages/django/db/models/fields/files.py", line 403, in __init__
    super().__init__(verbose_name, name, **kwargs)
  File "/home/eevel/wgtail310/lib/python3.10/site-packages/django/db/models/fields/files.py", line 243, in __init__
    self.storage = self.storage()
TypeError: get_rendition_storage() missing 1 required positional argument: 'instance'
`

Reply:
Alright, updated get_rendition_storage to return default_storage when an instance isn't available, and to call instance.get_storage() when one is provided.  The instance method checks for a custom setting, which may be overridden in subclasses, if desired.  The test case asserts that both get_rendition_storage(instance) and instance.get_storage() return the expected results.